### PR TITLE
RPM spec: Download source from github.com/linux-rdma/infiniband-diags

### PR DESCRIPTION
**Pull request by cyrilleverrier**
Create the RPM file from source files at github.com/linux-rdma/infiniband-diags instead of http://www.openfabrics.org because the link on http://www.openfabrics.org is dead.

Once this PR is merged, can one create a git tag "2.2.0" to make a new release with all fixes since last year?

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -8,7 +8,7 @@ AM_CONFIG_HEADER(config.h)
 AM_INIT_AUTOMAKE([subdir-objects])
 
 AC_SUBST(RELEASE, ${RELEASE:-unknown})
-AC_SUBST(TARBALL, ${TARBALL:-${PACKAGE}-${VERSION}.tar.gz})
+AC_SUBST(TARBALL, ${TARBALL:-${VERSION}.zip})
```
Comment (weiny2) [inline configure.ac]: Why remove ${PACKAGE}?

Right now changing this breaks being able to build directly with a make dist command.

I'm not familiar with autoconf enough to know if make dist can generated a zip file?

Would you happen to know?

Comment (cyrilleverrier) [reply configure.ac]: Hi @weiny2 

> Right now changing this breaks being able to build directly with a make dist command.

Could you please detail me how it breaks "make dist" ?

With my modifications, I tested it with:
```
./autogen.sh
./configure
make dist => infiniband-diags-2.1.0.tar.gz file is created
spectool -gf infiniband-diags/infiniband-diags.spec => 2.1.0.zip file is downloaded from https://github.com/linux-rdma/infiniband-diags/archive/2.1.0.zip
```

Comment (weiny2) [reply configure.ac]: Sorry I was not clear.  Make dist is fine but it outputs a tar.gz not a zip file.  So the spec file within the tar ball can't find the zip file and I get the following error when I run:

./autogen.sh && ./configure && make dist && rpmbuild -ta infiniband-diags-2.1.0.tar.gz
...
error: Bad source: /home/iweiny/dev/management/infiniband-diags/2.1.0.zip: No such file or directory

Basically the output from make dist is no longer able to be used to build with rpmbuild...  Perhaps that is old school and we should not care?

I'm not familiar with spectool but it looks like you need to have access to github for it to run?  I can probably test that but not sure I will have time today.

Comment (cyrilleverrier) [reply configure.ac]: Thank you for explaining your workflow.

> Basically the output from make dist is no longer able to be used to build with rpmbuild... Perhaps that is old school and we should not care?
Sorry for that, I fixed my previous commit by reverting the definition of the `TARBALL` macro in `configure.ac` to its original definition and modified the `Source` in the spec.in file to download the tar.gz file from github.com

> I'm not familiar with spectool but it looks like you need to have access to github for it to run?

Indeed, `spectool` can download the tarball pointed by the `Source` URL in the `infiniband-diags.spec` file

Tested with commit b125dd3:
1. Compile and build RPM from local source
```
./autogen.sh && ./configure && make dist && rpmbuild -ta infiniband-diags-2.1.0.tar.gz`
```

2. Download source from github and build RPM (only the generated infiniband-diags.spec is needed)
```
spectool -g -R infiniband-diags.spec
Getting https://github.com/linux-rdma/infiniband-diags/releases/download/2.1.0/infiniband-diags-2.1.0.tar.gz to /root/rpmbuild/SOURCES/infiniband-diags-2.1.0.tar.gz

rpmbuild -ba infiniband-diags.spec
``` 

3. RPM metadata
```
rpm --query --info -p infiniband-diags-2.1.0-1.el7.centos.x86_64.rpm 
Name        : infiniband-diags
Version     : 2.2.0
Release     : 1.el7.centos
Architecture: x86_64
Install Date: (not installed)
Group       : System Environment/Libraries
Size        : 1021735
License     : GPLv2 or BSD
Signature   : (none)
Source RPM  : infiniband-diags-2.2.0-1.el7.centos.src.rpm
Build Date  : Wed 24 Apr 2019 12:10:14 CEST
Build Host  : 4aad8109b704
Relocations : (not relocatable)
URL         : https://github.com/linux-rdma/infiniband-diags/
Summary     : OpenFabrics Alliance InfiniBand Diagnostic Tools
Description :
This package provides IB diagnostic programs and scripts needed to diagnose an
IB subnet.  infiniband-diags now also provides libibmad.  libibmad provides low
layer IB functions for use by the IB diagnostic and management programs. These
include MAD, SA, SMP, and other basic IB functions.
```

**Comment by jgunthorpe** [comment]
Squash and force push this thing please

**Comment by cyrilleverrier** [comment]
Commits have been squashed

**Pull request merged**